### PR TITLE
ADD - danger link button #ANDROID-14534

### PR DESCRIPTION
**Pull request by juangardi21**
### :goal_net: What's the goal?
Add danger link button, both small and inverse variants. Also small link and inverse small link variants added.

### :construction: How do we do it?
* _Create new button styles_
* _Update catalog_

### ☑️ Checks
- [x] I updated the documentation, including readmes and wikis. If this is a breaking change, update [UPGRADING.md](../UPGRADING.md) to inform users how to proceed. If no updates are necessary, indicate so.
- [x] Tested with dark mode.
- [x] Tested with API 21.

### :test_tube: How can I test this?
- [x] 🖼️ Screenshots/Videos

https://github.com/Telefonica/mistica-android/assets/76561300/b87644a1-5b70-4524-810a-8fdff8dc8b91

- [x] Mistica App QR or download link https://install.appcenter.ms/orgs/tuenti-organization/apps/mistica-catalog/distribution_groups/public/releases/458
- [x] Reviewed by Mistica design team

### CODE REVIEW

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/Buttons.kt [hunk 1]
```diff
@@ -64,7 +64,7 @@ private fun Buttons(
         )
 
         val scope = rememberCoroutineScope()
-        ButtonStyle.values()
+        ButtonStyle.entries
```
Comment (juangardi21) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/Buttons.kt:67]: Fix a warning

File: library/src/main/java/com/telefonica/mistica/compose/button/Button.kt [hunk 2]
```diff
@@ -215,7 +215,13 @@ enum class ButtonStyle {
     PRIMARY_SMALL_INVERSE,
     SECONDARY_INVERSE,
     SECONDARY_SMALL_INVERSE,
-    LINK_INVERSE
+    LINK_INVERSE,
```
Comment (juangardi21) [inline library/src/main/java/com/telefonica/mistica/compose/button/Button.kt:218]: New styles created

File: library/src/test/java/com/telefonica/mistica/button/ButtonTest.kt [hunk 3]
```diff
@@ -130,4 +130,10 @@ private fun ButtonStyle.getButtonLayout(): Int = when (this) {
     ButtonStyle.SECONDARY_INVERSE -> R.layout.secondary_button_inverse
     ButtonStyle.SECONDARY_SMALL_INVERSE -> R.layout.secondary_button_inverse_small
     ButtonStyle.LINK_INVERSE -> R.layout.link_button_inverse
+    ButtonStyle.LINK_SMALL -> R.layout.link_small_button
```
Comment (juangardi21) [inline library/src/test/java/com/telefonica/mistica/button/ButtonTest.kt:133]: New layouts for each new button style

File: library/src/main/res/layout/danger_link_button.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (yamal-alm) [inline library/src/main/res/layout/danger_link_button.xml:1]: these are the files used by ButtonTest right? They should be moved to library-utils module instead

Comment (juangardi21) [reply library/src/main/res/layout/danger_link_button.xml]: You are right, done

**Review by yamal-alm — APPROVED** [review]
nice job 👏

**Review by DevPabloGarcia — APPROVED** [review]
Nice Job!

**Review by dpastor — APPROVED** [review]
👍 Good job

**Review by dpastor — APPROVED** [review]

**Review by DevPabloGarcia — APPROVED** [review]

File: library/src/main/java/com/telefonica/mistica/compose/button/Button.kt [hunk 5]
```diff
@@ -271,6 +277,19 @@ fun LinkWithChevronPreview() {
     }
 }
 
+@Preview
+@Composable
+fun DangerLinkWithChevronPreview() {
+    MisticaTheme(brand = MovistarBrand) {
+        Button(
+            text = "Text",
+            withChevron = true,
```
Comment (aweell) [inline library/src/main/java/com/telefonica/mistica/compose/button/Button.kt:286]: buttonLinkDanger should not allow chevron config:

https://www.figma.com/file/koROdh3HpEPG2O8jG52Emh/%F0%9F%94%B8-Buttons-Component-Specs?type=design&node-id=238%3A6303&mode=design&t=EowkPiv1VvYzhQGr-1

Comment (juangardi21) [reply library/src/main/java/com/telefonica/mistica/compose/button/Button.kt:286]: You are right. Default value of Button.withChevron is false, so I'm going to remove this parameter on this Preview.

**Review by aweell — COMMENTED** [review]
Screenshot 2024-04-18 at 16 05 16

Disabled background in all brands | dark for inverse danger buttons shows an unexpected color, the top example is in blau but happens also the others.
Expected:

Screenshot 2024-04-18 at 16 07 39

The token buttonLinkDangerBackgroundInverse is almost transparent in dark for all brands.

There's an error in the TU brand tokens, that need to be solved in the tokens JSON file in mistica-design

https://github.com/Telefonica/mistica-design/pull/1737

**Review by aweell — APPROVED** [review]

**Pull request merged**